### ISSUE TITLE
Manage your SAP Java buildpack dependencies using BoM

### ISSUE THREAD

**Issue by nenaraab** [SAP/cloud-security-xsuaa-integration#400]
Manage your SAP Java buildpack dependencies using [Bill of Materials](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/6c6936e8e4ea40c9a9a69f6783b1e978.html)

SAP Java Buildpack sample:
https://github.com/SAP/cloud-security-xsuaa-integration/tree/master/samples/sap-java-buildpack-api-usage

See also here:
https://mvnrepository.com/artifact/com.sap.cloud.sjb.cf/sap-java-buildpack-bom

BLOCKED: sjb bom do not yet provide java-api!

![image](https://user-images.githubusercontent.com/5861301/97164402-5984e100-1782-11eb-9248-b21a9200bf36.png)

**Comment by nenaraab** [comment]
Hi @liga-oz 
it seems, that also the released version should have all required java-api and token-client lib dependencies.
If this is true, we are unblocked.
Best regards,
Nena

**Comment by liga-oz** [comment]
@nenaraab,

in Maven central for sap-java-buildpack-bom 1.28.1 I see only api module included, with 1.29.1 java-api, token-client and api are included, but this version is not yet released in maven central.

**Comment by MatKuhr** [comment]
Hi @liga-oz @nenaraab, this issue indicates to me that you plan on:

-  setting the `java-api`, `token-client` and `api` of `com.sap.xs.security` to  `provided` 
-  in all of your `pom.xml` files
- Once the SJB is released to maven central.

Is that correct or did I misinterpret this thread?

**Comment by nenaraab** [comment]
Hi @MatKuhr (cc @liga-oz ),

no, right now, this is only the case for applications that make use of SAP Java Buildpack!

The SJB is provided by SAP Java Buildpack and as of version 1.29.1 this will also include `java-api` and `token-client` client lib, as SAP Java Buildpack internally uses the implementation (`java-security` and the api versions has to be kept in synch). The version of SJB dependency shall match the version of the buildpack used.

Best regards,
Nena

**Comment by MatKuhr** [comment]
Could you clarify what `SJB` stands for? I assumed `SJB == SAP Java Buildpack`.

So that means you expect users of the XSUAA library to use exclusions like this when using the SAP Java Buildpack:

```xml
<dependency>
    <groupId>com.sap.cloud.security.xsuaa</groupId>
    <artifactId>token-client</artifactId>
    <exclusions>
        <exclusion>
            <groupId>java-api</groupId>
            <groupId>com.sap.cloud.security</groupId>
        </exclusion>
    </exclusions>
</dependency>
<dependency>
    <groupId>com.sap.cloud.security</groupId>
    <artifactId>java-api</artifactId>
    <scope>provided</scope>
</dependency>
```

**Comment by nenaraab** [comment]
Hi @MatKuhr 

yes, sjb stands probably for SAP Java Buildpack and is managed by them.
 
Yes, SAP Java buildpack users have to set "java-api" and as of SAP Java Buildpack version `1.29.1` "token-client" to `<scope>provided</scope>. As explained here:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/java-security/Migration_SAPJavaBuildpackProjects_V2.md
> We are going to refer to SJB as soon as the SJB includes "java-api" AND "token-client" (next to "api").

> Changing `java-api` to provided as part of `token-client` lib would be an incompatible change. 

Would that be a feasible solution?

Best regards,
Nena

**Comment by nenaraab** [comment]
Hi @liga-oz 

I think we can leverage the bom now in our SAP Java Buildpack sample :-)
https://github.com/SAP/cloud-security-xsuaa-integration/tree/master/samples/sap-java-buildpack-api-usage

https://mvnrepository.com/artifact/com.sap.cloud.sjb.cf/sap-java-buildpack-bom/1.31.1
- com.sap.cloud.security » java-api 2.8.0 
- com.sap.cloud.security.xsuaa » api 2.8.0
- com.sap.cloud.security.xsuaa » token-client 2.8.0

Best regards,
Nena

**Comment by liga-oz** [comment]
available since 2.8.3
